src/react-components/avatar-editor.js

```diff
@@ -0,0 +1,247 @@
+    const avatar = await (personalAvatarId
+      ? this.getAvatar(personalAvatarId)
+      : this.createOrUpdateAvatar({ name: "My Avatar", parent_avatar_id: BOT_PARENT_AVATAR }));
+
+    this.props.store.update({
+      profile: {
+        personalAvatarId: avatar.avatar_id
+      }
+    });
+
+    this.props.onAvatarChanged(avatar.avatar_id);
+
+    this.setState({ ...this.state, avatar });
+  };
+
+  getAvatar = avatarId =>
+    fetch(getReticulumFetchUrl(`/api/v1/avatars/${avatarId}`), {
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+        authorization: `bearer ${this.props.store.state.credentials.token}`
+      }
+    })
+      .then(r => r.json())
+      .then(({ avatars }) => avatars[0]);
+
+  createOrUpdateAvatar = avatar =>
+    fetch(getReticulumFetchUrl(avatar.avatar_id ? `/api/v1/avatars/${avatar.avatar_id}` : "/api/v1/avatars"), {
+      method: avatar.avatar_id ? "PUT" : "POST",
+      body: JSON.stringify({ avatar }),
+      headers: {
+        "Content-Type": "application/json",
+        authorization: `bearer ${this.props.store.state.credentials.token}`
+      }
+    })
+      .then(r => r.json())
+      .then(({ avatars }) => avatars[0]);
+
+  uploadAvatar = async e => {
+    e.preventDefault();
+
+    if (this.inputFiles.glb) {
+      const gltfLoader = new THREE.GLTFLoader();
+      const gltfUrl = URL.createObjectURL(this.inputFiles.glb);
+      const onProgress = console.log;
+      const parser = await new Promise((resolve, reject) =>
+        gltfLoader.createParser(gltfUrl, resolve, onProgress, reject)
+      );
+      URL.revokeObjectURL(gltfUrl);
+
+      const { content, body } = parser.extensions.KHR_binary_glTF;
+      this.inputFiles.gltf = new File([content], "file.gltf", {
+        type: "model/gltf"
+      });
+      this.inputFiles.bin = new File([body], "file.bin", {
+        type: "application/octet-stream"
+      });
+    }
+
+    const filesToUpload = ["gltf", "bin", "base_map", "emissive_map", "normal_map", "orm_map"].filter(
+      k => !!this.inputFiles[k]
+    );
+
+    this.setState({ uploading: true });
+
+    const fileUploads = await Promise.all(filesToUpload.map(f => this.inputFiles[f] && upload(this.inputFiles[f])));
+    const avatar = {
+      ...this.state.avatar,
+      files: fileUploads
+        .map((resp, i) => [filesToUpload[i], [resp.file_id, resp.meta.access_token, resp.meta.promotion_token]])
+        .reduce((o, [k, v]) => ({ ...o, [k]: v }), {})
+    };
+
+    await this.createOrUpdateAvatar(avatar);
+
+    this.props.onAvatarChanged(avatar.avatar_id);
+
+    this.setState({ uploading: false });
+    this.props.saveStateAndFinish();
+  };
+
+  fileField = (name, label, accept, disabled = false, title) => (
+    <div className={styles.fileInputRow} key={name} title={title}>
+      <label htmlFor={`avatar-file_${name}`}>
+        <div className="img-box">{this.state.avatar.files[name] && <img src={this.state.avatar.files[name]} />}</div>
```

Comment:
should use style imports for classes

Reply:
Usually only do that for the root element of a component, since they end up being scoped anyway, though this was just sloppy.